internal/provider/resource_database.go

```diff
@@ -140,29 +144,33 @@ func resourceDatabaseCreate(ctx context.Context, d *schema.ResourceData, meta in
 	name := d.Get("name").(string)
 	keyspace := d.Get("keyspace").(string)
 	cloudProvider := d.Get("cloud_provider").(string)
-	region := d.Get("region").(string)
+	regions := d.Get("region").([] interface{})
 
-	// Temporarily  disable region check
-	regionsResp, err := client.ListServerlessRegionsWithResponse(ctx)
-	if err != nil {
-		return diag.FromErr(err)
-	} else if regionsResp.StatusCode() != 200 {
-		return diag.Errorf("unexpected list available regions response: %s", string(regionsResp.Body))
+	if len(regions) < 1 {
+		return diag.Errorf("\"region\" array must have at least 1 region specified")
 	}
-	dbRegion := findMatchingRegion(cloudProvider, region, "serverless", *regionsResp.JSON200)
-	if dbRegion == nil {
-		return diag.Errorf("cloud provider and region combination not available: %s/%s", cloudProvider, region)
+
+	// Make sure all regions are valid
+	if err := ensureValidRegions(ctx, client, d); err != nil {
```

Comment:
Minor nit.  We use abbreviated for most other var names, should we consider identifying the `data` instead of `d` for  readability?  At first glance I wasn't sure what was being passed around.

Reply:
I can. I was just copying the function signatures used elsewhere, but it would make it easier to read.